### PR TITLE
setup jsx-a11y plugin

### LINKED ISSUE #18: Consider integrating eslint-plugin-jsx-a11y
https://github.com/evcohen/eslint-plugin-jsx-a11y might be a good addition to the `react` ruleset.

### PR DESCRIPTION
**Pull request by petrhanak**
Fixes #18.

### CODE REVIEW

**Comment by robertrossmann** [comment]
Thanks @petrhanak for the PR! When we discussed this addition with @dannytce we came to a conclusion that we cannot use their recommended ruleset because they set all the rules to `error`. I would like to keep that level to rules which detect broken, completely anti-pattern or otherwise extremely bad code. The rules in this plugin represent recommended best practices, therefore the rules should all be set to `warn`.

Could you please add the rules individually with their levels reduced down to `warn`? Thanks!

**Comment by petrhanak** [comment]
fixed `error` => `warn` and also added comments of what rules mean

**Review by robertrossmann — APPROVED** [review]
Very nice! Melikes. ❤️

**Comment by robertrossmann** [comment]
Waiting for additional review from @dannytce before merging.

**Comment by robertrossmann** [comment]
Idea: Would it make sense to split the rules into yet another configuration file instead of adding it to the _optional.js_ ? Something like _accessibility.js_ in the _react_ family.

What do you think @petrhanak @dannytce?

File: environments/react/accessibility.js
Comment (dannytce) [inline environments/react/accessibility.js]: Let's have this one as an error, because it breaks Safari

File: environments/react/accessibility.js
Comment (dannytce) [inline environments/react/accessibility.js:188]: Let's have this one as an error. It's a good practise to enforce developers to do that.

**Review by dannytce — CHANGES_REQUESTED** [review]

File: environments/react/accessibility.js
Comment (robertrossmann) [reply environments/react/accessibility.js:188]: If it's not absolutely terrible practice I would prefer keeping this at warn level. Best practices should not cause a failure in your test/deployment pipeline.

**Comment by robertrossmann** [comment]
@dannytce Are you okay with the changes made and with my comment on your review?

@petrhanak Some last things that need be done on this before we can (finally!) merge:

- Please update the main README.md's _Available rulesets_ section
- Please update the _environments/react/README.md_ with a more detailed information about the new ruleset
- Once done, please squash all the commits into a single commit and rebase on current master, there are some conflicts

Thank you for your efforts on this!

**Comment by robertrossmann** [comment]
Thanks, merged! I will release this together with #22 (once that's merged, too).

🍻 🎉 👍

**Pull request merged**
